website/addons/wiki/views.py

```diff
@@ -274,6 +287,8 @@ def project_wiki_view(auth, wname, path=None, **kwargs):
         if wiki_key not in node.wiki_pages_current and wiki_key != 'home':
             raise WIKI_PAGE_NOT_FOUND_ERROR
         if 'edit' in request.args:
+            if wiki_settings.is_publicly_editable:
+                raise HTTPError(http.UNAUTHORIZED)
```

Comment:
These are the errors that occur if you don't necessarily have permission to edit the wiki, but put 'edit' in your request. 
This specific case is if the wiki is publicly editable, but you are not logged in (unauthorized).
In all other cases, such an action is forbidden.